### PR TITLE
Improve FAE metric for DDP

### LINKED ISSUE #1792: Distributed tests fail
Recently while working on `ignite.contrib.metrics.regression` tests, I was trying to add distributed tests to some metrics that didn't have distributed tests. While i was running those tests when `world_size > 1` the result computed by the metric implemented in `ignite` differs from the one computed by `sklearn`(if imported from sklearn) / `numpy`. And here are examples for that:

- [Test wave hedges distance](https://gist.github.com/KickItLikeShika/497e80b2ebf2b99a9775e095868b62af)
`_test_distrib_integration()` fails in case of `world_size > 1` and if i didn't comment `all_gather()` in `_test_distrib_compute()` it fails as well!

- [Test fractional absolute error](https://gist.github.com/KickItLikeShika/8b6919f08b4d5ba531a363580517f791)
The same failures in the same two tests `_test_distrib_compute()` and `_test_distrib_integration()`.

related: #1284

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Fixes #1792 

Improving `FractionalAbsoluteError` implementation to be compatible with DDP and add the necessary tests

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/contrib/metrics/regression/fractional_absolute_error.py [hunk 1]
```diff
@@ -32,21 +33,26 @@ class FractionalAbsoluteError(_BaseRegression):
...
     """
 
+    @reinit__is_reduced
     def reset(self) -> None:
-        self._sum_of_errors = 0.0
+        self._sum_of_errors = torch.tensor(0.0, device=self._device)
         self._num_examples = 0
 
     def _update(self, output: Tuple[torch.Tensor, torch.Tensor]) -> None:
         y_pred, y = output
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/fractional_absolute_error.py:47]:
```suggestion
        y_pred, y = output[0].detach(), output[1].detach()
```

**Comment by sdesrozis** [comment]
@KickItLikeShika Thank you ! Looks good, just a comment.

File: ignite/contrib/metrics/regression/fractional_absolute_error.py [hunk 2]
```diff
@@ -32,21 +33,26 @@ class FractionalAbsoluteError(_BaseRegression):
         device: specifies which device updates are accumulated on. Setting the
             metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
             non-blocking. By default, CPU.
+
+    .. versionchanged:: 0.4.4
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/fractional_absolute_error.py:37]:
```suggestion
    .. versionchanged:: 0.5.0
```

**Review by sdesrozis — APPROVED** [review]
LGTM

**Review by vfdev-5 — APPROVED** [review]
Looks good, thanks @KickItLikeShika !
